rmgweb/database/views.py

```diff
@@ -590,6 +1039,284 @@ def parseSoluteDataComment(comment):
+    """
+    Returns a dictionary with solute parameter data for a given list of solute SMILES.
+    If a solvent is selected, solvation properties are also given.
+    """
+    solute_smiles_list = solute_smiles.split()
+
+    solute_data_results = {'Input SMILES': [],
+                           'Error': [],
+                           'E': [],
+                           'S': [],
+                           'A': [],
+                           'B': [],
+                           'L': [],
+                           'V': [],
+                           'Comment': []}
+
+    solute_data_list = []
+    solute_data_found_list = []  # a list of Boolean for whether the solute parameters E, S, A, B, L are all found. (not V)
+    additional_info_list = []  # a list of strings containing additional information to display in the result page
+
+    database.load('solvation')
+    db = database.get_solvation_database('', '')
+
+    # Add epistemic uncertainty columns for SoluteML model
+    if solute_estimator == 'SoluteML':
+        for solute_param_name in ['E', 'S', 'A', 'B', 'L']:
+            solute_data_results[f'{solute_param_name} epi. unc.'] = []
+        additional_info_list.append('epi. unc.: epistemic uncertainty of the SoluteML model.')
+    elif solute_estimator == 'SoluteGC':
+        additional_info_list.append('Comment: functional groups used to estimate the solute parameters')
+
+    # get predictions for each given SMILES
+    for smiles in solute_smiles_list:
+        # initialize the parameter values
+        Error = '-'
+        Comment = '-'
+        solute_data = SoluteData()
+        try:
+            solute_spc = Species().from_smiles(smiles)
+            solute_spc.generate_resonance_structures()
+        except:
+            solute_spc = None
+            Error = 'Unable to parse the SMILES'
+        # Get predictions using the selected method
+        if solute_estimator == 'expt':
+            if solute_spc is not None:
+                data = db.get_solute_data_from_library(solute_spc, db.libraries['solute'])
+                if data is not None:
+                    solute_data = data[0]
+                    Comment = f'From RMG-database solute library: {data[2].index}. {data[2].label}'
+                else:
+                    Comment = 'Not found in RMG-database'
+        elif solute_estimator == 'SoluteGC':
+            if solute_spc is not None:
+                try:
+                    solute_data = db.get_solute_data(solute_spc, skip_library=True)
+                    Comment = solute_data.comment
+                except:
+                    Error = 'Unable to get the prediction from the SoluteGC method'
+        elif solute_estimator == 'SoluteML':
+            solute_epi_unc_dict = {}
+            try:
+                avg_pre, epi_unc, valid_indices = SoluteML_estimator([[smiles]])
+                solute_data.E = avg_pre[0][0]
```

Comment:
You might be able to rewrite this block pythonically. I'm not 100% sure on the implementation but I'm imagining something like
```
[solute_data.E, solute_data.S, solute_data.A, solute_data.B, solute_data.L] = (avg_pre[0][i] for i in range(5)) 
```
Same for below, but double check that this works though.

Reply:
Thanks. I did it in this commit f82f20f in a new function `get_solute_data_from_SoluteML`.